crates/pipeline-manager/src/runner/local_runner.rs

```diff
@@ -127,33 +116,194 @@ pub async fn fetch_binary_ref(
+    ) -> (
+        oneshot::Sender<()>,
+        JoinHandle<mpsc::Receiver<mpsc::Sender<LogMessage>>>,
+    ) {
+        let (terminate_sender, mut terminate_receiver) = oneshot::channel::<()>();
+        let join_handle = spawn(async move {
+            // Buffered line readers such that we can already segment it into lines,
+            // such that stdout and stderr can be interleaved without making it unreadable
+            let mut lines_stdout = BufReader::new(stdout).lines();
+            let mut lines_stderr = BufReader::new(stderr).lines();
+
+            // Buffer with the latest lines
+            let mut logs = LogsBuffer::new(
+                Self::LOGS_BUFFER_LIMIT_BYTE,
+                Self::LOGS_BUFFER_LIMIT_NUM_LINES,
+            );
+
+            // All parties interested in receiving the logs
+            let mut log_followers: Vec<mpsc::Sender<LogMessage>> = Vec::new();
+
+            // Continues running until the logging is sent a termination message,
+            // or an I/O or send error is encountered
+            let mut stdout_finished = false;
+            let mut stderr_finished = false;
+            loop {
+                select! {
+                    // Terminate
+                    _ = &mut terminate_receiver => {
+                        Self::end_log_of_followers(&mut log_followers, LogMessage::End("LOG STREAM END: pipeline is being shutdown".to_string())).await;
+                        debug!("Terminating logging by request");
+                        break;
+                    }
+                    // New follower
+                    follower = log_follow_request_receiver.recv() => {
+                        if let Some(follower) = follower {
+                            Self::catch_up_and_add_follower(&mut logs, &mut log_followers, follower).await;
+                        } else {
+                            // The automata cannot recover from this as this is the only method to receive log followers from the runner.
+                            // Operational logging should not be active when the pipeline is shutdown, and only if the pipeline is
+                            // shutdown should it be deleted (which would cause this). As such, this is an error.
+                            error!("Log request channel closed during operational logging. Logs for this pipeline will no longer work.");
+                            break;
+                        }
+                    }
+                    // New stdout line
+                    line = lines_stdout.next_line(), if !stdout_finished => {
+                        match line {
+                            Ok(line) => match line {
+                                None => {
+                                    stdout_finished = true;
+                                }
+                                Some(line) => {
+                                    println!("{line}"); // Also print it to manager's stdout
```

Comment:
I guess we're printing unconditionally because pipeline's output is already controlled by the pipeline's log level.

Reply:
Yes, we can see as we go if we need to tune down the logging verbosity, or make this a runner configuration option